File: 0011-container-with-most-water/0011-container-with-most-water.java
class Solution {
	public int maxArea(int[] heights) {
		int maxiArea = 0, left = 0, right = heights.length - 1;
    int breadth, minHeight, area;
		
		while (left < right) {
			breadth = right - left;
			minHeight = Math.min(heights[left], heights[right]);
			area = breadth * minHeight;
			maxiArea = Math.max(maxiArea, area);
			if (heights[left] > heights[right]) {
				right--;
			} else {
				left++;
			}
		}
		return maxiArea;
	}
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
      maxi_area = left = 0
      right = len(height) - 1
        
      while left < right:
        breadth = right - left
        current_height = min(height[left], height[right])
        area = breadth * current_height
        maxi_area = max(area, maxi_area)
          
        if height[left] > height[right]:
          right -= 1
        else: 
          left += 1
        
      return maxi_area
        